pkg/schedule/schedule.go

```diff
@@ -149,17 +156,29 @@ func (f *fifo) run() {
 				f.mu.Unlock()
 				// clean up pending jobs
 				for _, todo := range pendings {
-					todo(f.ctx)
+					f.executeJob(todo, true)
 				}
 				return
 			}
 		} else {
-			todo(f.ctx)
+			f.executeJob(todo, false)
+		}
+	}
+}
+
+func (f *fifo) executeJob(todo Job, ctxDone bool) {
```

Comment:
nit: Instead of `ctxDone` I would call it `updatedFinishedStats` as we should talk about meaning of the parameter, and not why we had 'set' it.